x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer.js

```diff
@@ -1186,6 +1253,42 @@ export class TimeSeriesExplorer extends React.Component {
                 />
               );
             })}
+            {actualPlotFunction && (
+              <EuiFlexItem style={{ textAlign: 'right' }}>
+                <EuiFormRow
+                  label={i18n.translate('xpack.ml.timeSeriesExplorer.metricPlotByOption', {
+                    defaultMessage: 'Function',
+                  })}
+                >
+                  <EuiSelect
+                    options={[
+                      {
+                        value: 'avg',
+                        text: i18n.translate('xpack.ml.timeSeriesExplorer.plotByAvgOptionLabel', {
+                          defaultMessage: 'average',
+                        }),
+                      },
+                      {
+                        value: 'min',
+                        text: i18n.translate('xpack.ml.timeSeriesExplorer.plotByMinOptionLabel', {
+                          defaultMessage: 'min',
+                        }),
+                      },
+                      {
+                        value: 'max',
+                        text: i18n.translate('xpack.ml.timeSeriesExplorer.plotByMaxOptionLabel', {
+                          defaultMessage: 'max',
+                        }),
+                      },
+                    ]}
+                    value={actualPlotFunction ?? 'avg'}
+                    onChange={(e) => this.setPlotByFunction(e.target.value)}
+                    aria-label="Pick function to plot by (min, max, or average) if metric function"
+                  />
+                </EuiFormRow>
+              </EuiFlexItem>
+            )}
```

Comment:
Timeseriesexplorer component is already huge, consider moving this part.

Reply:
Moved to a separate file here https://github.com/elastic/kibana/pull/81662/commits/21bb6cb31d31959d096b318b15f2426c840d6ac9